core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,19 +135,35 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var deleteEmptyElements = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
             $scope.showAddItemButton();
           };
 
           $scope.showAddItemButton = function() {
+            deleteEmptyElements();
             $scope.isAddItemButtonPresent = true;
           };
 
           $scope.hideAddItemButton = function() {
             $scope.isAddItemButtonPresent = false;
           };
 
+          if ($scope.itemSchema().value === 'setInput') {
```

Comment:
Nope, sorry. You can't just add bits and pieces like this to a general schema in order to satisfy local needs! That leads to unmaintainable code (which is what happens when each client tries to modify it a bit here and there for their own purposes).

Rather, the functionality here auto-adds a new value. You need to find a way to do this in the context of the set input extension, and not here. Think about the set input interaction as a specific client who wants to change the value of the form, and change the value there instead.